### PR TITLE
Added cypress tests

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
The old assignment description from https://github.com/joinpursuit/axios_assignment/blob/master/axios_exercise.md is quite a bit different in wording than the other assignments. I wasn't sure how much of that I should change so I tried to minimally touch what I needed to make this a testable async/await lab.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,89 @@
...
+
+Create a `button` tag in your HTML.
+When you click on this `button`, you should fire your Axios request to draw five cards.
+Save them to a variable in your project.
+When you click the button again, you should draw five different cards.
+
+### Rendering Cards to the DOM
+
+When we click, we should see the images of five cards, in a row, underneath our button.
+This should be familiar to you from our Dog API image rendering.
```
Comment (mikeboyle) [inline README.md:65]: We will probably fix these directions in house (you are right that it is very different and IMO hard to follow). Just noting for myself that this reference to Dog API image rendering is probably out of date and should be replaced with specific instructions.

File: README.md [hunk 2]
```diff
@@ -1,19 +1,89 @@
...
+In your button event listener, after your Axios request, loop through your five cards.
+Create `img` tags for each of them, give each a class of `.image`, set each `src` to a the card's image URL, and append them to the DOM underneath your `button` tag.
+
+Just like our Dog API project, when you click the button again, it should remove/replace each card with a new card.
```
Comment (mikeboyle) [inline README.md:69]: Just flagging another dated Dog API reference -- no action required by you @JoshuaKGoldberg

File: README.md [hunk 3]
```diff
@@ -1,19 +1,89 @@
...
+
+Today, we're going to be working with the [Deck of Cards API](https://deckofcardsapi.com/).
+It's a foundational skill for us, as programmers, to learn how any particular API works from some time testing it out, so take 15 minutes or so to look at the API documentation and play around with it in Postman.
+
+Today, we're going to create an app that does these things:
+
+- Creates and shuffles a new deck when the page loads.
+- Includes a button to draw some cards from that deck.
+- Displays those cards to the user.
+- When the user clicks the button again, draws another set of cards from that same deck.
```
Comment (mikeboyle) [inline README.md:25]: We will fix the instructions in house so no action required @JoshuaKGoldberg , just flagging that these are the key user stories to implement and we should highlight them much more. (And reduce the narrative and hints below.)

File: cypress/integration/lab.test.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+describe("Index", () => {
+  it("starts the select with a default value of 5", () => {
+    cy.visit("./index.html");
+    cy.get("select").should("have.value", 5);
+  });
+
+  it("shows five cards from the retrieved deck ID when the button is clicked", () => {
+    visitWithFirstDeck();
+
+    cy.get(".card")
```
Comment (mikeboyle) [inline cypress/integration/lab.test.js]: [The instructions above](https://github.com/joinpursuit/Pursuit-Core-Web-Async-Await-Lab/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R65) say to use `img` tags with a class of `.image` for each card. So I think either the instructions or the test should be updated so that the specs look for the class and/or element we tell them to use in the instructions. IMO `card` makes more sense as a class name but I really don't have a preference how you reconcile it.

File: README.md [hunk 5]
```diff
@@ -1,19 +1,89 @@
...
 ## Submission Guidelines
-  * When finished, commit and push your work.
-  * Make a pull request on github.
-  * Submit the link to your pull request on Canvas.
 
-[Additional instructions]
\ No newline at end of file
+- When finished, commit and push your work.
+- Make a pull request on github.
+- Submit the link to your pull request on Canvas.
```
Comment (mikeboyle) [inline README.md:96]: Do you think you can add a note or small section explaining that these tests use fixtures? 
- what are fixtures
- where to find the fixtures
- tips for working with test output when there are fixtures

File: README.md [hunk 6]
```diff
@@ -1,19 +1,89 @@
...
+Today, we're going to be working with the [Deck of Cards API](https://deckofcardsapi.com/).
+It's a foundational skill for us, as programmers, to learn how any particular API works from some time testing it out, so take 15 minutes or so to look at the API documentation and play around with it in Postman.
+
+Today, we're going to create an app that does these things:
+
+- Creates and shuffles a new deck when the page loads.
+- Includes a button to draw some cards from that deck.
+- Displays those cards to the user.
+- When the user clicks the button again, draws another set of cards from that same deck.
+
```
Comment (mikeboyle) [inline README.md:26]: Can you add a screen gif that shows your solution?

File: index.html [hunk 7]
```diff
@@ -1,3 +1,6 @@
 <!DOCTYPE html>
     <h1>Hello World</h1>
+    <script src="https://unpkg.com/axios/dist/axios.min.js"></script>
+    <script src="./index.js"></script>
+</body>
```
Comment (mikeboyle) [inline index.html:9]: Can you add the missing tags here? 
- opening html 
- opening and closing head tags with the script tags in there (ok to use `defer` for index.js)
- opening body

File: cypress/integration/lab.test.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+const clickForDeck = (fixture, count = 5) => {
+  cy.intercept(`https://deckofcardsapi.com/api/deck/${deckId}/draw/?count=${count}`, { fixture });
+  cy.get("button").click();
+};
+
+const visitWithFirstDeck = () => {
+  cy.intercept("https://deckofcardsapi.com/api/deck/new/shuffle/?deck_count=1", {
+    deck_id: deckId,
+  }).as("newDeck");
```
Comment (mikeboyle) [inline cypress/integration/lab.test.js]: How is the alias used here? Not blocking, just a question for my own education.

File: cypress/integration/lab.test.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  });
+
+  it("shows five cards from the retrieved deck ID when the button is clicked", () => {
+    visitWithFirstDeck();
+
+    cy.get(".card")
+      .should("have.length", 5)
+      .first()
+      .should("have.attr", "src", "https://deckofcardsapi.com/static/img/2D.png");
+  });
```
Comment (mikeboyle) [inline cypress/integration/lab.test.js:36]: A couple of things might slip through the cracks here that I think are worth testing (and are doable with the fixtures).
- When we fetch 5 cards, all 3 cards are the expected cards from the fixtures (not the same single card repeated 5 times)
- When we fetch 3 cards next time around, all 3 cards are the expected cards from the fixtures (not reused or repeated)

**Review by mikeboyle — COMMENTED** [review]
Looking great! Like you said, the narrative instructions are weird and our team will deal with those in house. Otherwise there are a few small comments.

After looking at your solution code, I don't think this lab will keep them busy for long. Do you have any ideas for a bonus feature we could add? Examples:
- add a button that removes all cards and replaces the deck with a new one
- click on a single card to replace it with a new one
- display something to indicate that the deck has been exhausted

File: cypress/integration/lab.test.js (hunk 8 above)
Comment (JoshuaKGoldberg) [reply cypress/integration/lab.test.js]: Ah, unused code, I'd forgotten to remove it.

File: README.md (hunk 5 above)
Comment (JoshuaKGoldberg) [reply README.md:96]: I'm not 100% that what I wrote will completely satisfy what you're looking for... but it's a start I suppose 😄

File: README.md (hunk 6 above)
Comment (JoshuaKGoldberg) [reply README.md:26]: Gif showing drawing 5 cards, then 4, then 1, then 5 again

**Comment by JoshuaKGoldberg** [comment]
Ooh, I like the "display remaining" suggestion. Will do!

**Comment by mikeboyle** [comment]
@JoshuaKGoldberg I suggest we get this PR merged (which I'm ready to do once the tests are passing on the solution branch). Then a new PR for the bonus task? This will also let us start shaping the instructional text in parallel. Sounds good?

**Comment by JoshuaKGoldberg** [comment]
Ah, I ended up doing the bonus as part of debugging the CI-only failures. But otherwise yes! 😄

**Comment by JoshuaKGoldberg** [comment]
All right @mikeboyle I ended up making the "remaining cards" display a part of the core assignment, and using the "click a card to fetch a new one" as the bonus. In retrospect this makes sense: the cards display doesn't require any new async/await behavior.

**Review by mikeboyle — APPROVED** [review]
LGTM! 👏  👏  👏  This one was a lot of work and I really appreciate you wrangling all of this together! I'll merge and me / the instructor team will move ahead with cleaning up the instructions language. Thank you! 🚀
